### PR TITLE
Wrong use of 'an' when you rename 'Organizer' group role inside 'Group Types' settings.

### LINKED ISSUE #602: Wrong use of 'an' when you rename 'Organizer' group role inside 'Group Types' settings.
**Describe the bug**
For 'Organizer' group role, you will see 'You're an Organizer' on Groups page. 
But if you rename the group role and write 'Teacher' or any other word, you will see 'You're an Teacher' which is wrong. 

**To Reproduce**
1. Go to BuddyBoss -> groups -> group types. 
2. Create a group type. 
3. Rename the 'Organizer' group role to 'Teacher'.
4. Assign a group to this 'Group Type'.
5. Go to 'Groups' directory page.
6. See the wrong text on the assigned group. 

**Screenshots**
https://prnt.sc/r7y4l1

**Support ticket links**
https://buddyboss.zendesk.com/agent/tickets/62562

### PR DESCRIPTION
**Pull request by KartikSuthar**
### All Submissions:

* [x] Have you followed the [Pull Requests Guidelines](https://github.com/buddyboss/buddyboss-platform/wiki/Submitting-Pull-Requests#pull-request-guidelines)?
* [x] Does your code follow the [WordPress' Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Fixes #602.

### How to test the changes in this Pull Request:
1. Go to BuddyBoss -> groups -> group types. 
2. Create a group type. 
3. Rename the 'Organizer' group role to 'Teacher'.
4. Assign a group to this 'Group Type'.
5. Go to 'Groups' directory page.
6. See the text on the assigned group. 

### Proof Screenshots or Video
http://somup.com/cYnu1thtbw

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

### CODE REVIEW

File: src/bp-groups/bp-groups-template.php [hunk 1]
```diff
@@ -7355,3 +7355,22 @@ function bp_get_group_current_invite_tab() {
...
+ * @return string
+ */
+function bp_get_article_prefix( $string ) {
+	if ( empty( $string ) ) {
+		return $string;
+	}
+
+	$vowelArry = array('a','e','i','o','u');  // array of vowel
+	$prefix = in_array( strtolower( substr( $string ,0,1 ) ), $vowelArry ) ? __( 'an', 'buddyboss' ) : __( 'a', 'buddyboss' );
+	return $prefix . ' ' . $string;
```
Comment (chetansatasiya) [inline src/bp-groups/bp-groups-template.php]:
```suggestion
	return apply_filters( 'bp_get_article_prefix ', $prefix . ' ' . $string );
```

File: src/bp-groups/bp-groups-template.php [hunk 2]
```diff
@@ -7355,3 +7355,22 @@ function bp_get_group_current_invite_tab() {
...
+ *
+ * @param string $string  Text string for adding the article prefix.
+ * @return string
+ */
+function bp_get_article_prefix( $string ) {
+	if ( empty( $string ) ) {
+		return $string;
+	}
+
+	$vowelArry = array('a','e','i','o','u');  // array of vowel
```
Comment (chetansatasiya) [inline src/bp-groups/bp-groups-template.php]:
```suggestion
	$vowel_array = array('a','e','i','o','u');  // array of vowel
```

File: src/bp-groups/bp-groups-template.php [hunk 3]
```diff
@@ -7355,3 +7355,22 @@ function bp_get_group_current_invite_tab() {
...
+	$prefix = in_array( strtolower( substr( $string ,0,1 ) ), $vowelArry ) ? __( 'an', 'buddyboss' ) : __( 'a', 'buddyboss' );
```
Comment (chetansatasiya) [inline src/bp-groups/bp-groups-template.php]:
```suggestion
	$prefix = in_array( strtolower( substr( $string ,0,1 ) ), $vowel_array ) ? __( 'an', 'buddyboss' ) : __( 'a', 'buddyboss' );
```

File: src/bp-groups/bp-groups-template.php (hunk 1 above)
Comment (chetansatasiya) [reply src/bp-groups/bp-groups-template.php]: add a filter so if anyone has to make any changes then they can easily.

Comment (KartikSuthar) [reply src/bp-groups/bp-groups-template.php]: Added filter as named `bp_get_article_prefix`.

File: src/bp-groups/bp-groups-template.php (hunk 2 above)
Comment (KartikSuthar) [reply src/bp-groups/bp-groups-template.php]: Changed variable name to `$vowel_array`

File: src/bp-groups/bp-groups-template.php (hunk 3 above)
Comment (KartikSuthar) [reply src/bp-groups/bp-groups-template.php]: Changed variable name to `$vowel_array`

**Comment by KartikSuthar** [comment]
@chetansatasiya Made above-listed changes and fix with code sniffer

**Comment by chetansatasiya** [comment]
@KartikSuthar I need @eisenwasser thoughts on this and then I will merge this PR.

**Comment by eisenwasser** [comment]
@KartikSuthar @chetansatasiya As discussed in scrum, we should limit this solution to English language(s) only.

**Comment by KartikSuthar** [comment]
@chetansatasiya @eisenwasser , Added support for the English language(s) only.

**Comment by eisenwasser** [comment]
@chetansatasiya Can you do the code review/ testing on this one?

**Comment by chetansatasiya** [comment]
> @chetansatasiya Can you do the code review/ testing on this one?

@eisenwasser I am testing

**Comment by chetansatasiya** [comment]
@eisenwasser @KartikSuthar see this https://prnt.sc/rae22b I think as we discussed @KartikSuthar we have to keep `a` always in other languages except `en`.

I am not sure @eisenwasser  can you confirm like this is correct?

Code looks good to me

**Comment by anoopmishrabuddyboss** [comment]
Tested it on my local with the latest `fix/602` branch issue is fixed. 
Screenshots:- with other languages https://prnt.sc/rbsmqq. With English https://prnt.sc/rbsnxo 
@chetansatasiya please review and merge to the dev branch.

**Comment by chetansatasiya** [comment]
@anoopmishrabuddyboss OR @ankitguptaindia I need you to test this.

**Comment by anoopmishrabuddyboss** [comment]
@chetansatasiya Already tested working fine.
Screenshots:- with other languages https://prnt.sc/rbsmqq. With English https://prnt.sc/rbsnxo

**Pull request merged**